### PR TITLE
#7 added acm optional certificate

### PR DESCRIPTION
**Pull request by powerwisdom**

### CODE REVIEW

**Comment by danial-k** [comment]
Thanks @powerwisdom , might be an idea to add an example without an ACM certificate to the readme.md file and mention as a new paragraph something to the effect of:
```
To avoid using an ACM certificate and use the default cloudfront certificate, set `cloudfront_default_certificate` to `true` and do not provide `alias` or `acm_certificate_arn`
```

**Comment by danial-k** [comment]
Also @powerwisdom , what are your thoughts on making the alias optional as well with a default of `null` like you've done for the `acm_certificate_arn`? That way a user that doesn't want to use ACM doesn't keep having to set `alias = []` or `alias = null`.

**Comment by powerwisdom** [comment]
> Thanks @powerwisdom , might be an idea to add an example without an ACM certificate to the readme.md file and mention as a new paragraph something to the effect of:
> 
> ```
> To avoid using an ACM certificate and use the default cloudfront certificate, set `cloudfront_default_certificate` to `true` and do not provide `alias` or `acm_certificate_arn`
> ```

Very good idea @danial-k will quickly get updating on this ticket now :)

**Comment by powerwisdom** [comment]
> 



> Also @powerwisdom , what are your thoughts on making the alias optional as well with a default of `null` like you've done for the `acm_certificate_arn`? That way a user that doesn't want to use ACM doesn't keep having to set `alias = []` or `alias = null`.

I think this is a brill idea @danial-k its also less hassle too. once we implement alias being optional, should I also update the readme?

File: README.md [hunk 1]
```diff
@@ -13,6 +13,8 @@ To make changes to a distribution once deployed, update the origin path.
 For example, a version 1.0 project could be placed in a `/v1.0` directory, and the next version of the project placed in `/v1.1`.
 Updating the `origin_path` from `/v1.0` to `/v1.1` would invalidate the cache and redeploy the new distribution.
 
+To avoid using an ACM certificate and use the default cloudfront certificate, set `cloudfront_default_certificate` to `true` and do not provide `alias` or `acm_certificate_arn`.
```
Comment (danial-k) [inline README.md:16]: Should we move this to its own example? So the first example would be `Cloudfront distribution with custom ACM certificate` and your new example would be `Cloudfront distribution with default certificate`?

File: README.md [hunk 2]
```diff
@@ -63,6 +65,17 @@ module "cloudfront" {
   origin_path                 = "/v1"
 }
 
+# Main cloudfront distribution with S3 origin with optional ACM certification and optional alias
+module "cloudfront" {
+  source                      = "voquis/cloudfront-s3-origin-acm/aws"
+  version                     = "0.0.3"
```
Comment (danial-k) [inline README.md]: Should we preempt to `0.0.5` :see_no_evil: 
Could we also update the other example?

File: README.md [hunk 3]
```diff
@@ -63,6 +65,17 @@ module "cloudfront" {
...
+  bucket_id                   = module.s3.s3_bucket.id
+  bucket_arn                  = module.s3.s3_bucket.arn
+  bucket_regional_domain_name = module.s3.s3_bucket.bucket_regional_domain_name
+  origin_path                 = "/examplefolder"
+  cloudfront_default_certificate = true
```
Comment (danial-k) [inline README.md]: Should we line these up to avoid terraform fmt complaining if someone copy/pastes.

File: README.md (hunk 2 above)
Comment (danial-k) [reply README.md]: We could also update the `module "acm"` from `0.0.2` to `0.0.4` :pray: https://github.com/voquis/terraform-aws-acm-dns-validation/tags

Comment (danial-k) [reply README.md]: Could we also update the `module "s3"` from `0.0.2` to `0.0.4`? :pray: 
https://github.com/voquis/terraform-aws-s3-encrypted/tags

File: main.tf [hunk 4]
```diff
@@ -113,6 +113,7 @@ resource "aws_cloudfront_distribution" "this" {
   # Viewer certificate
   viewer_certificate {
     acm_certificate_arn      = var.acm_certificate_arn
+    cloudfront_default_certificate = var.cloudfront_default_certificate
```
Comment (danial-k) [inline main.tf:116]: could we run `terraform fmt` inside the module to fix all formatting please?

**Review by danial-k — COMMENTED** [review]
Could we add the `.terraform.lock.hcl` to `.gitignore` so it's not included in commit please :pray:

**Review by danial-k — APPROVED** [review]

**Pull request merged**